twisted/names/test/test_rootresolve.py

```diff
@@ -209,20 +209,20 @@ def test_lookupChecksClass(self):
         badClass.CLASS = HS
         servers = {
             ('1.1.2.3', 53): {
-                ('foo.example.com', A): {
-                    'answers': [('foo.example.com', badClass)],
-                    'authority': [('foo.example.com', Record_NS('ns1.example.com'))],
-                    'additional': [('ns1.example.com', Record_A('10.0.0.2'))],
+                (b'foo.example.com', A): {
+                    'answers': [(b'foo.example.com', badClass)],
+                    'authority': [(b'foo.example.com', Record_NS(b'ns1.example.com'))],
+                    'additional': [(b'ns1.example.com', Record_A('10.0.0.2'))],
                 },
             },
             ('10.0.0.2', 53): {
-                ('foo.example.com', A): {
-                    'answers': [('foo.example.com', Record_A('10.0.0.3'))],
+                (b'foo.example.com', A): {
+                    'answers': [(b'foo.example.com', Record_A('10.0.0.3'))],
                 },
             },
         }
         resolver = self._getResolver(servers)
-        d = resolver.lookupAddress('foo.example.com')
+        d = resolver.lookupAddress(b'foo.example.com')
```

Comment:
[Not uniformly he didn't](https://github.com/twisted/twisted/commit/ce06c31fdfdbf4dafa77bbc4e94435d9b38a1211#diff-fcdce5901f50f49227b552b53e1bdcc1R503).

I'd say it's relevant only in that it indicates that previous patches have not addresses this concern either. =P I am willing for either @hawkowl or @glyph to tell me that we don't need to address this at this time, but as far as I understand it part of the porting work requires that we re-evaluate almost anything that is defined to take `str`.

To be clear, a totally acceptable resolution to this, IMO, would be to redefine the interface to take `bytes`: that's a no-op on Python 2, and if this module has not been completely ported may not be a compat break on Py 3.